api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -178,7 +188,12 @@ message Cluster {
     // Optional alternative to cluster name to present to EDS. This does not
     // have the same restrictions as cluster name, i.e. it may be arbitrary
     // length.
-    string service_name = 2;
+    string service_name = 2 [(udpa.annotations.field_migrate).oneof_promotion = "name_specifier"];
+
+    // Resource locator for EDS. This is mutually exclusive to *service_name*.
+    // [#not-implemented-hide:]
+    udpa.core.v1.ResourceLocator eds_resource_locator = 3
+        [(udpa.annotations.field_migrate).oneof_promotion = "name_specifier"];
```

Comment:
Don't we want this field to replace both the `service_name` field and the `eds_config` field?  In other words, I think we want either this new field or *both* `eds_config` and `service_name`.

Maybe the right thing to do here is to leave the `EdsClusterConfig` message alone and move this new field into a top-level field in the `Cluster` message.  Then the `oneof` can contain this new field and the `EdsClusterConfig` field.

Reply:
So, as per the resolution algorithm in the bootstrap, I think it's fine to have "peer config sources". This get checked first, then the bootstrap. It provides a more gradual path to adoption for folks who don't already provide config sources for everything in bootstrap.